File: ch4 supplemental/repl5.cs
using System;
class MainClass {
	public static void Main (string[] args) {
		Console.WriteLine("Do you have children? (Y/N)");
		
		int childCountAsInt =0;
		string hasChildren = Console.ReadLine();
		
		if(hasChildren == "Y" || hasChildren == "y"){
		
			Console.WriteLine("How many children?");
			string childCount = Console.ReadLine();
			
			if(!int.TryParse(childCount, out childCountAsInt)){
				Console.WriteLine("You entered an incorrect answer of '" + childCount + "', sorry.");
			}
		}
		
		Console.WriteLine("Your answer of " + childCountAsInt + " children has been recorded.");
		Console.WriteLine("Thanks for your participation, questionaire complete.");
	} 
}
      

File: ch4 supplemental/repl5.js
var hasChildren = prompt("Do you have children (Y/N)");
var childCount =0;

if(hasChildren == "Y" || hasChildren =="y") 
{
	var childCountPrompt = prompt("How many children?");
	
	if(parseInt(childCountPrompt)!=NaN && isFinite(childCountPrompt)){
		childCount = childCountPrompt;
	}
	
	else{
	console.log("You entered '" + childCount + "' which is an invalid response that can not be counted in the survey")
	}
	
}

console.log("Your survey result of " + childCount + " children has been recorded");

console.log("Thanks for your participation, questionnaire complete.")